tests/integration/json_test.go

```diff
@@ -0,0 +1,101 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odojsonoutput", func() {
+	var project string
+	var oc helper.OcRunner
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		oc = helper.NewOcRunner("oc")
+		project = oc.CreateRandProject()
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		oc.DeleteProject(project)
+		os.RemoveAll(".odo")
+	})
+
+	Context("odo machine readable output on empty project", func() {
+		// odo app list -o json
+		It("should be able to return empty list", func() {
+			actual := helper.CmdShouldPass("odo app list -o json --project " + project)
+			desired := `{"kind":"List","apiVersion":"odo.openshift.io/v1alpha1","metadata":{},"items":[]}`
+			Expect(desired).Should(MatchJSON(actual))
+			time.Sleep(5 * time.Second)
```

Comment:
why sleep?

Reply:
Yah, this is unfortunate. Just to avoid race condition i am using sleep for now. Actually in the very next scenario the deleted project is being captured as a project item. 

```
{
          "kind": "List",
          "apiVersion": "odo.openshift.io/v1alpha1",
          "metadata": {},
          "items": [
            {
              "kind": "Project",
              "apiVersion": "odo.openshift.io/v1alpha1",
              "metadata": {
                "name": "myproject",
                "creationTimestamp": null
              },
              "spec": {
                "apps": [
                  "app"
                ]
              },
              "status": {
                "active": true
              }
            },
            {
              "kind": "Project",
              "apiVersion": "odo.openshift.io/v1alpha1",
              "metadata": {
                "name": "project-qgqzfane",
                "creationTimestamp": null
              },
              "spec": {
                "apps": [
                  "app"
                ]
              },
              "status": {
                "active": false
              }
            },
            {
              "kind": "Project",
              "apiVersion": "odo.openshift.io/v1alpha1",
              "metadata": {
                "name": "project-qkyblqrz",
                "creationTimestamp": null
              },
              "spec": {
                "apps": [
                  "app"
                ]
              },
              "status": {
                "active": false
              }
            }
          ]
        }
```

Can you please suggest some better solution